src/benchmark/bbench.cc

```diff
@@ -0,0 +1,94 @@
+/*
+  This file is part of Leela Chess Zero.
+  Copyright (C) 2020 The LCZero Authors
+
+  Leela Chess is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  Leela Chess is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with Leela Chess.  If not, see <http://www.gnu.org/licenses/>.
+
+  Additional permission under GNU GPL version 3 section 7
+
+  If you modify this Program, or any covered work, by linking or
+  combining it with NVIDIA Corporation's libraries from the NVIDIA CUDA
+  Toolkit and the NVIDIA CUDA Deep Neural Network library (or a
+  modified version of those libraries), containing parts covered by the
+  terms of the respective license agreement, the licensors of this
+  Program grant you additional permission to convey the resulting work.
+*/
+
+#include "benchmark/bbench.h"
+
+#include "chess/board.h"
+#include "mcts/node.h"
+#include "neural/factory.h"
+#include "utils/optionsparser.h"
+
+namespace lczero {
+namespace {
+const int kDefaultThreads = 1;
+
+const OptionId kThreadsOptionId{"threads", "Threads",
+                                "Number of (CPU) worker threads to use.", 't'};
+const OptionId kBatchesId{"batches", "",
+                          "Number of batches to run as a benchmark."};
+const OptionId kMaxBatchSizeId{"max-batch-size", "",
+                               "Maximum batch size to benchmark."};
+const OptionId kFenId{"fen", "", "Benchmark initial position FEN."};
+}  // namespace
+
+void BackendBenchmark::Run() {
+  OptionsParser options;
+  NetworkFactory::PopulateOptions(&options);
+  options.Add<IntOption>(kThreadsOptionId, 1, 128) = kDefaultThreads;
+
+  options.Add<IntOption>(kBatchesId, 1, 999999999) = 100;
```

Comment:
This default value might be a bit high.  For a big net 10 seems fine and runs much faster (obviously).